pennylane/math/tensorbox.py

```diff
@@ -12,10 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """This module contains the TensorBox abstract base class."""
-# pylint: disable=import-outside-toplevel
+# pylint: disable=import-outside-toplevel,too-many-public-methods
 import abc
+import functools
+import numbers
+import sys
 
-import numpy as np
+
+def wrap_output(func):
+    """Decorator to automate the wrapping of TensorBox method outputs.
+
+    When applied to a TensorBox method, it inserts an additional argument
+    into the signature, ``wrap_output``. By default, this is True, causing
+    the output of the method to be wrapped as a TensorBox; specifying
+    ``wrap_output=False`` when calling the method results in the
+    underlying tensor itself being returned.
+    """
+
+    @functools.wraps(func)
+    def _wrapper(*args, **kwargs):
+        wrap = kwargs.pop("wrap_output", True)

```

Comment:
Yep, by default it is True! Should we change it to False?

Making it False by default increases the line length significantly, since you can no longer chain tensorbox methods without specifying `wrap_output=True` at every method.

With it being True by default, you only need to specify `wrap_output=False` on the _last_ method call.

Reply:
Ah I see now how it works!